### PR TITLE
`raise-coverage` skill

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
## Summary

Introduces the **`raise-coverage`** skill — a Spine-native agent skill that
raises JVM test coverage for a Gradle module or source path by localizing
uncovered lines/branches from Kover's JaCoCo-format XML report and generating
policy-compliant unit tests. Adjacent infrastructure (`KoverConfig`, the
JaCoCo → Kover deprecation set, the `migrate` Hugo prune block) is included
so the skill can run end-to-end against any current Spine consumer repo.

## The skill — `raise-coverage`

Triggered as `/raise-coverage <:module | path | --triage>`. Lives at
`.agents/skills/raise-coverage/` and propagates to all consumer repos via
`./config/pull`. Wired into Codex via `agents/openai.yaml` and into Claude
via `.claude/commands/raise-coverage.md`; listed in `_TOC.md` and registered
in the `migrate` Hugo-only prune block.

### Abilities

- **Step 0 — Kover gate.** Before doing anything else, inspects the repo's
  coverage setup and branches:
  - Kover applied everywhere → silently proceeds.
  - Nothing in place → silently installs Kover.
  - Vanilla JaCoCo in ≥ 1 module → emits a structured proposal
    (*Detected · Plan · Translation notes · Manual-review surfaces · Smoke
    check*) and **waits for explicit approval** before editing.
- **JaCoCo → Kover migration.** On approval, applies the repo-wide
  mechanical recipe in `references/migrate-to-kover.md`: swaps
  `JacocoConfig.applyTo` → `KoverConfig.applyTo`, drops `jacoco` from
  plugin blocks, rewrites CI / `.codecov.yml` / scripts to Kover task
  names and XML paths, and smoke-checks that `koverXmlReport` produces a
  non-empty JaCoCo-format XML at the expected path.
- **Gap localization.** Runs `:<module>:koverXmlReport`, parses the XML
  for `ci == 0` lines and `mb > 0` partial branches, drops generated /
  example / test paths, and discards **non-actionable** gaps the engine
  cannot credit even with a perfect test (`inline` / `inline reified`
  definitions; unreachable `require` / `check` / `error` guards;
  `throw helper(...)` where the helper throws internally — the Spine
  `Exceptions` idiom).
- **`--triage` mode.** Read-only: produces a ranked gap report across the
  repo (or a module) and stops without proposing or writing tests.
- **Proposal-before-write.** Builds a concrete test-case list per gap
  (method/branch, input, expected outcome, required stubs) and **waits
  for approval** before generating anything.
- **Test generation.** New tests are always written in **Kotlin** (JUnit
  Jupiter structure + Kotest assertions), regardless of whether the code
  under test is Kotlin or Java. Class names use the **`Spec`** suffix
  (e.g. `AbstractSourceFileSpec`). No mocking framework — stubs only.
- **Verification.** Re-runs `koverXmlReport`, confirms the previously
  uncovered lines/branches no longer appear, and that no `.codecov.yml`
  target regressed.

### Other notable changes

- **New `KoverConfig` aggregator** (`buildSrc/.../KoverConfig.kt`, +247
  LOC) — the live runtime successor to `JacocoConfig.applyTo(rootProject)`.
  Applies the Kover plugin at the root, wires
  `dependencies { kover(project(...)) }` for every Kover-enabled
  subproject, pins `useJacoco(version = Jacoco.version)`, and pushes the
  union of generated-class FQNs into Kover's exclude filter — preserving
  the generated-code filtering previously provided by `CodebaseFilter`.
- **JaCoCo helpers deprecated.** `JacocoConfig`, `CodebaseFilter`,
  `FileFilter`, `FileExtensions`, `FileExtension`, `PathMarker`,
  `TaskName`, plus the `jacoco-kotlin-jvm.gradle.kts` and
  `jacoco-kmm-jvm.gradle.kts` script plugins — marked `@Deprecated` with
  pointers to the Kover path. Kept on disk so existing consumers keep
  building; will be removed in a future release. The script plugins now
  emit `logger.warn(...)` when applied.
- **Dependency bumps.** 
  - Kover `0.9.1` → `0.9.8` (`Kover.kt` + `buildSrc/build.gradle.kts`); 
  - JaCoCo engine `0.8.13` → `0.8.14`  (`Jacoco.kt`).
- **`migrate` script.** Adds `raise-coverage` to the Hugo-only prune
  block so documentation-only repos do not receive the JVM-specific
  skill.
- **`scripts/upload-artifacts.sh`.** Adds a `DEPRECATED` comment above
  the `JACOCO_REPORTS=…` block pointing at the migration recipe and
  noting Kover does not expose raw `.exec` paths; copyright header
  refreshed. No behaviour change.
- **`AGENTS.md` — "Asking questions" section.** Codifies the one-question-
  per-message rule (with inline options for small option sets) and the
  preference for reasonable assumptions over additional questions when
  the answer would not materially change the next step.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/commands/raise-coverage.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- Target: $ARGUMENTS — a Gradle module (e.g. `:base`), a source path, or
+  `--triage` to only produce the ranked Kover gap report without generating tests.
+- First-time setup: the skill enforces Kover. If vanilla JaCoCo is found
+  anywhere, the skill proposes a repo-wide migration and **waits for your
+  approval**. See `.agents/skills/raise-coverage/references/migrate-to-kover.md`.
+- Order: localize gaps from Kover's JaCoCo-format XML → propose concrete test
+  cases and **wait for confirmation** → generate → re-run
+  `:<module>:koverXmlReport` to verify the gap closed.
+- Honor `.agents/testing.md` (stubs not mocks; Kotlin → Kotest, Java → Google
+  Truth) and `.agents/coding-guidelines.md` for Kotlin/Java idioms.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .claude/commands/raise-coverage.md]: Fixed in 56d269d7 — the wrapper now states the Kotlin-always + Spec policy in line with SKILL.md.

File: .agents/skills/raise-coverage/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+### Verify (smoke check)
+
+After applying, run `./gradlew :<module>:koverXmlReport --quiet` on the smallest
+leaf migrated JVM module; if the root was touched, also run
+`./gradlew koverXmlReport --quiet`. Confirm
+`<module>/build/reports/kover/report.xml` exists, is non-empty, and the first
+non-XML-declaration line contains `<report `. If a `DOCTYPE` is present, confirm
+it points at JaCoCo's `report.dtd` — that confirms `useJacoco(...)` is in
+effect. Failure → stop; do not fall through to the Workflow.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/SKILL.md]: Fixed in 56d269d7 — the smoke check now branches: JVM `:<module>:koverXmlReport` → `report.xml`; KMP `:<module>:koverXmlReportJvm` → `reportJvm.xml`.

File: .agents/skills/raise-coverage/agents/openai.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+interface:
+  display_name: "Raise Coverage"
+  short_description: "Migrate to Kover if needed, then generate unit tests to close coverage gaps."
+  default_prompt: "Use $raise-coverage. Step 0 first: detect the coverage setup. If vanilla JaCoCo is found anywhere, propose a one-shot repo-wide migration to Kover and wait for approval before editing. If no coverage frontend is in place, install Kover silently. After the smoke check (`./gradlew :<module>:koverXmlReport --quiet` produces a non-empty `build/reports/kover/report.xml`), run the normal flow: localize uncovered lines and branches from Kover's JaCoCo-format XML report, propose a test-case list and wait for approval, then generate policy-compliant unit tests (stubs not mocks; written in Kotlin with Kotest assertions regardless of whether the code under test is Kotlin or Java; class names use the `Spec` suffix, e.g. `AbstractSourceFileSpec`) and re-run `koverXmlReport` to verify the gaps are closed. Tests-only changes do not require a version bump."
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/agents/openai.yaml]: Fixed in 56d269d7 — the default_prompt now spells out the JVM (`koverXmlReport` / `report.xml`) vs KMP (`koverXmlReportJvm` / `reportJvm.xml`) smoke-check branches.

File: .agents/skills/raise-coverage/references/migrate-to-kover.md [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+Apply these edits to each module's `build.gradle.kts`:
+
+### Add Kover
+
+- If `buildSrc` is on the classpath and exposes the `Kover` object:
+  ```kotlin
+  plugins {
+      id(io.spine.dependency.test.Kover.id)
+  }
+  ```
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/references/migrate-to-kover.md]: Fixed in 56d269d7 — section 3 now uses the literal plugin id `id("org.jetbrains.kotlinx.kover")` (Gradle's Kotlin DSL `plugins { }` block does not accept references to buildSrc constants across the Gradle versions Spine supports). The `Kover.id` constant is mentioned as a comment so a reader can cross-check the source of truth.

File: .agents/skills/raise-coverage/references/migrate-to-kover.md [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+}
+
+dependencies {
+    kover(project(":foo"))
+    kover(project(":bar"))
+    // … one entry per consuming module
+}
+
+kover {
+    useJacoco(version = io.spine.dependency.test.Jacoco.version)
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/references/migrate-to-kover.md]: Fixed in 56d269d7 — section 4 long-form now uses literal values (`id("org.jetbrains.kotlinx.kover") version "0.9.8"` and `useJacoco(version = "0.8.14")`), since the Kotlin DSL `plugins { }` block requires literals. A comment points the reader at `io.spine.dependency.test.Kover` / `Jacoco` as the source of truth for those versions.

File: buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    private fun applyExcludes(sub: Project, fqns: Collection<String>) {
+        if (fqns.isEmpty()) {
+            return
+        }
+        sub.extensions.configure(KoverProjectExtension::class.java) {
+            reports {
+                filters {
+                    excludes {
+                        classes(fqns.toExclusionPatterns())
+                    }
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt:170]: Fixed in 56d269d7 — `useJacoco(Jacoco.version)` is now applied on each eligible subproject in addition to the root. The call is idempotent for `jvm-module` / `kmp-module` consumers (which already pin the same version) and matters for subprojects that apply Kover directly. Refactored `applyExcludes` → `configureSubproject` accordingly.

File: .claude/commands/raise-coverage.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .claude/commands/raise-coverage.md]: Fixed in 56d269d7 (same fix as the Copilot thread on this line) — the wrapper now states the Kotlin-always + Spec policy in line with SKILL.md.

File: .agents/skills/raise-coverage/references/migrate-to-kover.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+  }
+  ```
+
+### Strip JaCoCo
+
+- Remove `jacoco` from `plugins { }` (or the `id("jacoco")` line, or
+  `apply<JacocoPlugin>()`, or `apply(plugin = "jacoco")`).
+- Replace `apply(plugin = "jacoco-kotlin-jvm")` / `apply(plugin = "jacoco-kmm-jvm")`
+  with `id("jvm-module")` / `id("kmp-module")` when that is the module's role;
+  otherwise drop and add `id(Kover.id)` directly.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/references/migrate-to-kover.md]: Fixed in b1583e6d — replaced `id(Kover.id)` with the literal `id("org.jetbrains.kotlinx.kover")` in section *Strip JaCoCo* and section 6 *KMP recipe*, with a comment pointing at `Kover.id` as the source of truth. Also tightened the section-6 code block to use the literal JaCoCo version. The remaining `useJacoco(version = Jacoco.version)` references are all inside `kover { }` (or KDoc for `KoverConfig`) blocks, where the Kotlin DSL accepts buildSrc references.

File: .agents/skills/raise-coverage/references/coverage-signals.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+### Extracting gaps for a class
+
+The XML carries a `DOCTYPE` pointing at a public DTD, so always pass `--nonet` to
+`xmllint` (or use the Python recipe) — parsing must never reach the network. The
+report has no XML namespace, so the XPath is plain:
+
+```bash
+xmllint --nonet \
+  --xpath '//class[@name="io/spine/.../MyType"]//line[@ci="0" or @mb > 0]' \
+  <module>/build/reports/kover/report.xml
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/references/coverage-signals.md:119]: Fixed in fe427380 — replaced the buggy XPath. JaCoCo/Kover stores `<line>` under `<sourcefile>` (sibling of `<class>` under `<package>`), not under `<class>`. The new snippet queries by `//package[@name=...]/sourcefile[@name=...]/line[...]` for line-level gaps, with a separate `//class[...]/method[counter[@type="BRANCH" and @missed>0]]/@name` query for branch-method localization. The Python recipe was already correct (it traverses `<sourcefile>`).

File: .claude/commands/raise-coverage.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+---
+description: Ensure the repo is on Kover (migrate from JaCoCo if needed), then localize coverage gaps and generate missing unit tests for a module or path.
+argument-hint: "<:module | path | --triage>"
+allowed-tools: Read, Edit, Write, Grep, Glob, Bash(./gradlew:*), Bash(git status:*), Bash(find:*)
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .claude/commands/raise-coverage.md]: Fixed in fe427380 — added `Bash(xmllint:*)` and `Bash(python3:*)` to `allowed-tools`, matching the XML-parsing recipes documented in `references/coverage-signals.md`.

File: .agents/tasks/buildsrc-gradle-review-findings.md [hunk 10]
```diff
@@ -75,6 +75,10 @@ For each, add `group = SpineTaskGroup.name` and an imperative
       (`updateGitHubPages`, `copyJavadocDocs`, `copyHtmlDocs`,
       `updatePagesTask`).
 - [x] `buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/JacocoConfig.kt:165, 196`
+- [ ] **Superseded by Kover-only migration**: this file is deprecated; do not
+      invest in micro-rewrites. See
+      `.agents/skills/raise-coverage/references/migrate-to-kover.md`.
+      `buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/JacocoConfig.kt:165, 196`
       (`jacocoRootReport`, `copyReports`).
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/tasks/buildsrc-gradle-review-findings.md]: Fixed in fe427380 — folded the supersede note from the A.2 duplicate into the existing `[x]` entry, and reshaped the other four standalone supersede annotations (sections B, C, F, G) the same way: the original work item is now the `[x]` bullet with the supersede note appended. No duplicate checklist items remain.

**Comment by alexander-yevsyukov** [comment]
@codex review

File: .agents/skills/raise-coverage/SKILL.md [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+
+### Verify (smoke check)
+
+Pick the smallest migrated leaf module and run its Kover report task, then
+inspect the produced XML:
+
+- **JVM module** — task `:<module>:koverXmlReport`; XML at
+  `<module>/build/reports/kover/report.xml`.
+- **KMP module** — task `:<module>:koverXmlReportJvm`; XML at
+  `<module>/build/reports/kover/reportJvm.xml`. (The skill only migrates the
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/SKILL.md]: Fixed in 9d20fa27. Verified against `buildSrc/src/main/kotlin/kmp-module.gradle.kts:181-190` — only `kover { reports { total { … } } }` is configured, with no named `variant("…") { … }` block, so Kover does not generate `koverXmlReport<Variant>` tasks. Updated SKILL.md, openai.yaml, coverage-signals.md, migrate-to-kover.md, both planning docs, and the deprecated `jacoco-kmm-jvm` header to point at `:<module>:koverXmlReport` and `build/reports/kover/report.xml` on KMP too. The two remaining mentions of `koverXmlReportJvm` in the docs explicitly negate its existence in Spine's configuration.

**Comment by alexander-yevsyukov** [comment]
@codex review

File: buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+                "In a single-module Gradle project, `KoverConfig` is NOT needed. " +
+                        "Apply the Kover plugin directly to the module instead."
+            }
+            project.pluginManager.apply(Kover.id)
+            KoverConfig(project).configure()
+        }
+    }
+
+    private fun configure() {
+        rootProject.gradle.projectsEvaluated {
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt]: Fixed in 446793b4. `configure()` no longer wraps everything in `gradle.projectsEvaluated`:

- Root extension is configured eagerly (`configureRoot()`), so `useJacoco(...)` and the `total.xml.onCheck` flag are set at apply time.
- Per-subproject hooks now fire inside `pluginManager.withPlugin(Kover.id) { … }`, which runs the moment a subproject applies Kover — either immediately (already-applied subprojects) or later in the same configuration phase. Either way, the callback runs **before** Kover's own `afterEvaluate` finalization, so the per-subproject `useJacoco(...)`, the `kover(project(...))` dependency, and the exclude filter are all visible when Kover builds its task graph.
- Class KDoc rewritten to describe the new timing model and to drop the misleading "uses `gradle.projectsEvaluated` internally" line.

File: buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+     *
+     * Returns an empty list if the project does not expose a Java/Kotlin
+     * `main` source set (for example, a pure non-JVM module).
+     */
+    private fun generatedClassFqns(project: Project): List<String> {
+        val main = project.sourceSets.findByName(SourceSet.MAIN_SOURCE_SET_NAME)
+            ?: return emptyList()
+        return generatedSrcDirs(main)
+            .asSequence()
+            .filter { it.exists() && it.isDirectory }
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/io/spine/gradle/report/coverage/KoverConfig.kt:233]: Fixed in 446793b4. Both per-subproject and root excludes are now wrapped in a lazy `Provider<Iterable<String>>` (`perSubprojectExcludePatternsProvider` / `generatedExcludePatternsProvider`) and passed to Kover's `classes(Provider<Iterable<String>>)` overload. The `generated/` directory walk — and the `it.exists() && it.isDirectory` filter inside `generatedClassFqns(...)` — only runs when the provider is queried at task-graph time, which is **after** `protoc` and the other code-generation tasks have produced their outputs. On a clean build the generated dirs are visible by then, so they no longer get dropped before the FQN list is built.

File: .agents/tasks/raise-coverage.md [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+
+1. **`.agents/skills/raise-coverage/SKILL.md`** — frontmatter `name` +
+   `description: >`. Sections: Goal/scope (JaCoCo-only; human `src/main` only) ·
+   Inputs (`$ARGUMENTS` = `:module` | path | `--triage`) · Workflow (1 resolve
+   target → 2 localize gaps w/ JaCoCo → 3 read code-under-test + existing tests +
+   collaborators → 4 **propose test-case list and WAIT**; `--triage` stops here
+   with the ranked report → 5 generate → 6 verify) · Test-generation rules (stubs
+   not mocks; Kotlin = JUnit5 + Kotest, Java = JUnit5 + Truth; proto →
+   truth-proto-extension; cover edge cases; scaffold `when`/sealed branches; skip
+   generated/excluded paths) · Report format · Safety (never weaken a
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/tasks/raise-coverage.md]: Refreshed deliverable #1 in 446793b4 to match the current shape of the skill: Kover-only (using its JaCoCo-format XML report), Step 0 ensure-Kover gate with the `--triage` read-only carve-out, **Kotlin + Kotest universal** (written in Kotlin even when the code under test is Java; `truth-proto-extension` reachable only as an isolated fallback for Protobuf assertions Kotest cannot express), and the `Spec`-suffix naming convention.

File: .agents/skills/raise-coverage/SKILL.md [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+
+Run this **before** the Workflow below. Branch on the repo's current coverage
+setup (detection patterns and full migration recipe in
+[`references/migrate-to-kover.md`](references/migrate-to-kover.md)):
+
+1. **Kover applied everywhere already** — silently proceed to the Workflow.
+2. **No coverage plugin anywhere** — silently install Kover (per the recipe).
+   Record "Migration: installed Kover" in the final Report. No approval gate
+   for this branch.
+3. **Vanilla JaCoCo in ≥1 module** (with or without Kover alongside) — emit a
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/SKILL.md:92]: Fixed in 446793b4. Step 0 now branches on `$ARGUMENTS`: under `--triage` it is contractually read-only — if Kover is not applied everywhere, the skill emits a "Setup required" report (lists the affected modules, points at `references/migrate-to-kover.md`, asks the user to re-run without `--triage`) and **stops without writing or proposing**. The "silent install" branch (no coverage plugin in place) and the "propose-and-wait" branch (vanilla JaCoCo detected) only run when `--triage` is not requested. The Safety section also gained an explicit "`--triage` is read-only" bullet so the invariant cannot drift again.

**Comment by alexander-yevsyukov** [comment]
@codex review

File: .agents/skills/raise-coverage/references/migrate-to-kover.md [hunk 16]
```diff
@@ -0,0 +1,350 @@
...
+  blocks. See §4 (root aggregation) for the long-form equivalent if `buildSrc`
+  is not on the classpath.
+- **Lifecycle gotcha — do not call `KoverConfig.applyTo(...)` from inside
+  `gradle.projectsEvaluated { … }`.** Many Spine consumer repos wrap
+  `JacocoConfig.applyTo(project)` in that block; carrying the pattern over
+  fails with `Cannot run Project.afterEvaluate(Action) when the project is
+  already evaluated`, because Kover's plugin registers its own `afterEvaluate`
+  hooks at apply time. Lift the call to top level in the root build script;
+  `KoverConfig.configure()` already uses `gradle.projectsEvaluated` internally
+  for subproject discovery, so deferred behaviour is preserved.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/references/migrate-to-kover.md]: Fixed in 089d114c. The migration note no longer says `KoverConfig.configure()` uses `gradle.projectsEvaluated`. It now documents the current lifecycle: call `KoverConfig.applyTo(rootProject)` at top level; the root is configured eagerly, and subprojects are discovered through `pluginManager.withPlugin(...)` callbacks before Kover finalizes its reports.

File: .agents/skills/raise-coverage/SKILL.md [hunk 17]
```diff
@@ -0,0 +1,240 @@
...
+     branches (`mb > 0`). Prioritize methods whose `BRANCH` counter has
+     `missed > 0`.
+   - Drop any class under an excluded path (generated / examples / test).
+   - Discard **non-actionable** gaps the engine cannot credit even with a
+     perfect test (see `references/coverage-signals.md`): Kotlin `inline` /
+     `inline reified` functions (their bytecode is inlined into each call
+     site, so the definition lines stay `ci=0` regardless of tests) and
+     unreachable guards (`require`/`check`/`error` paths the public API
+     cannot trigger). Report these as non-actionable instead of proposing
+     tests for them.
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply .agents/skills/raise-coverage/SKILL.md]: Fixed in 089d114c. `SKILL.md` now includes `throw helper(...)` lines where the helper throws internally in the non-actionable gap list, matching `coverage-signals.md`, so the skill reports those gaps instead of proposing tests for them.

**Comment by alexander-yevsyukov** [comment]
@codex review

**Comment by alexander-yevsyukov** [comment]
@codex review

**Review by armiol — APPROVED** [review]
